src/error.rs

```diff
@@ -119,7 +119,7 @@ pub enum Error {
     #[error("Wallet file is missing")]
     WalletFileMissing,
     /// Wrong wallet password
-    #[error("Wrong password")]
+    #[error("Invalid password, reached the maximum number of attempts")]
```

Comment:
I would probably rename this as well.

This let me think that we might need a different kind of error.
It looks like `InvalidPassword` is actually a conversion error for the `BlockModeError`. So I would do the follow:

1. Change the current `InvalidPassword(#[from] block_modes::BlockModeError),` to reflect what is actually doing (a wrapper to `BlockModeError`, such as we have for IO error)
2. Add a new Error that is returned when we have finished the attempts, to be used in both `recovery` and `password` (that would remove the necessity to have `bail!`) with a message that is good for both.

Basically this:
```rust
    #[error("Invalid password, reached the maximum number of attempts")]
    InvalidPassword(#[from] block_modes::BlockModeError),
```
become:
```rust
    #[error("Block Mode Error")]
    BlockMode(#[from] block_modes::BlockModeError),
```
(It's fine if it's generic, we never use directly anyway, it's just a proxy)

Then we add a new error such as:
```rust
    /// Reached the maximum number of attempts
    #[error("Reached the maximum number of attempts")]
    AttemptsExhausted,
```

And we use it in both location, for the password:
```rust
                    Err(_) if attempt > 2 => {
                        return Err(Error::AttemptsExhausted)?
                    }
                    Err(_) => {
                        println!("Invalid password please try again");
                        attempt += 1;
                    }
```

And for the recovery:
```rust
            Err(err) if attempt > 2 => match err.downcast_ref::<ErrorKind>() {
                Some(ErrorKind::InvalidWord) => {
                    return Err(Error::AttemptsExhausted)?
                }
                _ => return Err(err),
            },
            Err(_) => {
                println!("Invalid recovery phrase please try again");
                attempt += 1;
            }
```

Reply:
way better !